imageregistry/v1/types.go

```diff
@@ -303,8 +302,65 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
+	// +optional
+	NetworkAccess *AzureNetworkAccess `json:"networkAccess,omitempty"`
+}
+
+type AzureNetworkAccess struct {
+	// type is the network access level to be used for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// When type: Internal is used, a vnetName, subNetName and privateEndpointName
+	// may optionally be specified. If unspecificed, the image registry operator
+	// will discover vnet and subnet names, and generate a privateEndpointName.
+	// Defaults to "External".
+	// +kubebuilder:default:="External"
+	Type AzureNetworkAccessType `json:"type,omitempty"`
+	// internal defines the vnet and subnet names to configure a private
+	// endpoint and connect it to the storage account in order to make it
+	// private.
+	// when type: Internal and internal is unset, the image registry operator
+	// will discover vnet and subnet names, and generate a private endpoint
+	// name.
+	// +optional
+	Internal *AzureNetworkAccessInternal `json:"internal,omitempty"`
+}
+
+type AzureNetworkAccessInternal struct {
+	// vnetName is the name of the vnet the registry operates in.
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:MinLength=2
+	// +kubebuilder:validation:Pattern=`^[0-9A-Za-z][0-9A-Za-z_.-]*[0-9A-Za-z_]$`
```

Comment:
Validations like this must be spelled out in prose within the godoc. These validations are non-obvious and not exposed to users within the generated docs. You'll want something along the lines of
```
// The vnetName must be between 2 and 64 characters in length and must consist only of
// alphanumeric characters, hyphens (-), periods (.) and underscores (_).
// It must start with an alphanumeric character and end with an alphanumeric character or an underscore.
```

Reply:
Done.